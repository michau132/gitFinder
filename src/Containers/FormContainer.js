import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import FormView from '../Components/Form/FormView';
import takeUserNameAndFetchData from '../actions/fetchData';


const FormContainer = ({ onFormSubmit, history }) => (
  <FormView
    onFormSubmit={onFormSubmit}
    historyPush={history.push}
  />
);

const mapDispatchToProps = dispatch => ({
  onFormSubmit: val => dispatch(takeUserNameAndFetchData(val)),
});

FormContainer.propTypes = {
  onFormSubmit: PropTypes.func.isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired,
  }).isRequired,
};

export default withRouter(connect(null, mapDispatchToProps)(FormContainer));
