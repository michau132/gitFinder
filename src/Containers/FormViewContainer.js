import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import FormView from '../Components/Form/FormView';
import takeUserNameAndFetchData from '../actions/fetchData';


const FormViewContainer = ({onFormSubmit, history }) => {
  
  return <FormView onFormSubmit={onFormSubmit} historyPush={history.push} />
};

const mapDispatchToProps = dispatch => ({
  onFormSubmit: val => dispatch(takeUserNameAndFetchData(val)),
});

FormViewContainer.propTypes = {
  onFormSubmit: PropTypes.func.isRequired,
};

export default withRouter(connect(null, mapDispatchToProps)(FormViewContainer));
