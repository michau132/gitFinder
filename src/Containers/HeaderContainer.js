import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import githubUsernameRegex from 'github-username-regex';
import Header from '../Components/Header';
import takeUserNameAndFetchData from '../actions/fetchData';

class HeaderContainer extends Component {
  static propTypes = {
    onFormSubmit: PropTypes.func.isRequired,
    history: PropTypes.shape({
      push: PropTypes.func.isRequired,
    }).isRequired,
  }

  constructor(props, context) {
    super(props, context);
    this.state = {
      inputValue: '',
      errorInput: null,
    };
  }

  onFormSubmit = (e) => {
    const { onFormSubmit, history } = this.props;
    const { inputValue, errorInput } = this.state;
    e.preventDefault();
    if (errorInput) {
      return;
    }
    onFormSubmit(inputValue);
    history.push(`/${inputValue}`);
  }

  updateInputValue = (val) => {
    const { value } = val.target;
    if (githubUsernameRegex.test(value)) {
      this.setState({
        inputValue: value,
        errorInput: null,
      });
    } else {
      this.setState({
        errorInput: true,
        inputValue: value,
      });
    }
  }

  render() {
    const { inputValue, errorInput } = this.state;
    return (
      <Header
        onFormSubmit={this.onFormSubmit}
        updateInputValue={this.updateInputValue}
        inputValue={inputValue}
        errorInput={errorInput}
      />
    );
  }
}


const mapDispatchToProps = {
  onFormSubmit: takeUserNameAndFetchData,
};

export default withRouter(connect(null, mapDispatchToProps)(HeaderContainer));
