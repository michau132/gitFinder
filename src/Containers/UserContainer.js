import React, { Component, Fragment } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import takeUserNameAndFetchData from '../actions/fetchData';
import LoaderHOC from '../hoc/LoaderHOC';
import ErrorHOC from '../hoc/ErrorHOC';

class UserContainer extends Component {
  static propTypes = {
    fetchData: PropTypes.func.isRequired,
    match: PropTypes.shape({
      params: PropTypes.shape({
        user: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
    user: PropTypes.shape({
      userInfo: PropTypes.shape({
        login: PropTypes.string.isRequired,
      }).isRequired,
    }).isRequired,
  }

  constructor(props, context) {
    super(props, context);
    this.state = {};
  }

  componentDidMount() {
    const { fetchData, match, user } = this.props;
    if (user.userInfo.login.length === 0) {
      fetchData(match.params.user);
    }
  }

  componentDidUpdate(prevProps) {
    const { fetchData, match } = this.props;
    const previousUser = prevProps.match.params.user;
    const nextUser = match.params.user;
    if (previousUser !== nextUser) {
      fetchData(match.params.user);
    }
  }

  render() {
    const { render } = this.props;
    return (
      <Fragment>
        {
          render({ ...this.props })
        }
      </Fragment>
    );
  }
}

const mapStateToProps = state => ({
  user: state,
});

const mapDispatchToProps = {
  fetchData: takeUserNameAndFetchData,
};

const UserContainerWithLoading = ErrorHOC(LoaderHOC(UserContainer));

export { UserContainer };

export default
withRouter(
  connect(mapStateToProps, mapDispatchToProps)(UserContainerWithLoading),
);
