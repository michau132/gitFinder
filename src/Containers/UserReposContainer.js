import React from 'react';
import { connect } from 'react-redux';
import {
  setReposOnKeyUp,
  selectUserRepo,
  openSelectedRepos,
  hideSelectedRepos,
  hideSingleRepo,
  showAllRepos,
  selectAllRepos,
} from '../actions';
import openSingleRepo from '../actions/openSingleRepo';
import UserRepos from '../Components/UserRepos';

const UserReposContainer = props => <UserRepos {...props} />;

const mapDispatchToProps = {
  setReposOnKeyUp,
  selectUserRepo,
  openSelectedRepos,
  hideSelectedRepos,
  showAllRepos,
  hideSingleRepo,
  openSingleRepo,
  selectAllRepos,
};

export { UserReposContainer };

export default connect(null, mapDispatchToProps)(UserReposContainer);
