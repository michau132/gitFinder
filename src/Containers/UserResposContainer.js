import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { uploadFilterReposData } from '../actions';
import filterOnKeyUp from '../actions/filterOnKeyUp';
import selectUserRepo from '../actions/selectUserRepo';
import openSelectedRepos from '../actions/openSelectedRepos';
import hideSelectedRepos from '../actions/hideSelectedRepos';
import showAllRepos from '../actions/showAllRepos';
import hideSingleRepo from '../actions/hideSingleRepo';
import openSingleRepo from '../actions/openSingleRepo';
import UserRepos from '../Components/UserRepos';

class UserResposContainer extends Component {
  static propTypes = {
    uploadFilterReposData: PropTypes.func.isRequired,
    userRepos: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      forks_count: PropTypes.number,
      stargazers_count: PropTypes.number,
      description: PropTypes.string,
      html_url: PropTypes.string.isRequired,
    })).isRequired,
  }

  componentDidMount() {
    // eslint-disable-next-line no-shadow
    const { uploadFilterReposData, userRepos } = this.props;
    uploadFilterReposData(userRepos);
  }

  render() {
    return (
      <UserRepos {...this.props} />
    );
  }
}

const mapStateToProps = state => ({
  userRepos: state.user.userRepos,
  filterRepos: state.filterRepos,
});

const mapDispatchToProps = dispatch => ({
  uploadFilterReposData: repos => dispatch(uploadFilterReposData(repos)),
  filterOnKeyUp: event => dispatch(filterOnKeyUp(event)),
  selectUserRepo: id => () => dispatch(selectUserRepo(id)),
  openSelectedRepos: e => dispatch(openSelectedRepos(e)),
  hideSelectedRepos: e => dispatch(hideSelectedRepos(e)),
  showAllRepos: e => dispatch(showAllRepos(e)),
  hideSingleRepo: id => () => dispatch(hideSingleRepo(id)),
  openSingleRepo: id => () => dispatch(openSingleRepo(id)),
});

export default connect(mapStateToProps, mapDispatchToProps)(UserResposContainer);
