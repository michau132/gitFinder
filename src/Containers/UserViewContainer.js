import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import UserView from '../Components/User/UserView';
import takeUserNameAndFetchData from '../actions/fetchData';

class UserViewContainer extends Component {
  componentDidMount() {
    const { onValuePassedThroughParams, match } = this.props;
    const { user } = match.params;
    onValuePassedThroughParams(user);
  }

  render() {
    const { user } = this.props;
    return (
      <UserView user={user} />
    );
  }
}

const mapStateToProps = state => ({
  user: state.user,
});

const mapDispatchToProps = dispatch => ({
  onValuePassedThroughParams: val => dispatch(takeUserNameAndFetchData(val)),
});

UserViewContainer.propTypes = {
  user: PropTypes.shape({
    userInfo: PropTypes.shape({
      login: PropTypes.string.isRequired,
      email: PropTypes.string,
      avatar_url: PropTypes.string,
      name: PropTypes.string,
    }).isRequired,
    userRepos: PropTypes.arrayOf(
      PropTypes.shape({
        name: PropTypes.string.isRequired,
        pushed_at: PropTypes.string.isRequired,
      }),
    ).isRequired,
  }).isRequired,
  onValuePassedThroughParams: PropTypes.func.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      user: PropTypes.string.isRequired,
    }).isRequired,
  }).isRequired,
};

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(UserViewContainer));
