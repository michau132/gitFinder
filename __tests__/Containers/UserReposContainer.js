import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import data from 'fake-data';
import { UserReposContainer } from '../../src/Containers/UserReposContainer';

describe('testing UserReposContainer', () => {
  let wrapper;
  let props;
  beforeEach(() => {
    props = {
      ...data.__getAll,
      selectUserRepo: jest.fn(),
      hideSingleRepo: jest.fn(),
      openSingleRepo: jest.fn(),
    };
    wrapper = shallow(<UserReposContainer {...props} />);
  });
  test('renders without crashing', () => {
    expect(wrapper).toBeDefined();
  });

  test('should match snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
