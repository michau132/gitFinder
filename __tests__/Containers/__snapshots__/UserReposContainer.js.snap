// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing UserReposContainer should match snapshot 1`] = `
<UserRepos
  hideSingleRepo={[MockFunction]}
  openSingleRepo={[MockFunction]}
  selectUserRepo={[MockFunction]}
  userInfo={
    Object {
      "avatar_url": "https://via.placeholder.com/150",
      "email": "aoko123@gmail.com",
      "loctation": "Raszyn",
      "login": "aOko123",
      "name": "Adam Oko",
    }
  }
  userRepos={
    Array [
      Object {
        "description": "First repo description",
        "forks_count": 0,
        "html_url": "https://github.com/aOko123/firstRepo",
        "id": 1,
        "isChecked": false,
        "isFounded": false,
        "isHidden": false,
        "name": "firstRepo",
        "stargazers_count": 0,
      },
      Object {
        "description": "Second repo description",
        "forks_count": 0,
        "html_url": "https://github.com/aOko123/secondRepo",
        "id": 2,
        "isChecked": false,
        "isFounded": false,
        "isHidden": false,
        "name": "secondRepo",
        "stargazers_count": 0,
      },
    ]
  }
/>
`;
